lib/ts/fetch.ts

```diff
@@ -281,19 +294,17 @@ export default class AuthHttpRequest {
                 } else {
                     logDebugMessage("doRequest: rid header was already there in request");
                 }
+                await setTokenHeadersIfRequired(clonedHeaders);
 
                 logDebugMessage("doRequest: Making user's http call");
                 let response = await httpCall(configWithAntiCsrf);
                 logDebugMessage("doRequest: User's http call ended");
 
-                const idRefreshToken = response.headers.get("id-refresh-token");
-                if (idRefreshToken) {
-                    logDebugMessage("doRequest: Setting sIRTFrontend: " + idRefreshToken);
-                    await setIdRefreshToken(idRefreshToken, response.status);
-                }
+                await saveTokensFromHeaders(response);
```

Comment:
Earlier we used to only store the front-token and anti-csrf if it was a non 401 response, but now we do this regardless - does this have any side effect?

Reply:
@porcellus missed this?

Reply:
No, I just wanted to discuss this on the call, but we didn't end up doing it. (Also, this is basically the same issue as answered in another comment thread about saving tokens)
Basically: since the access token and front-token values are tied together, their updates should be close together as well. Otherwise, the front end could get into an inconsistent state.
This might not be an issue here, but it still seems better to always save these updates if the backend sends them. There could be some edge cases, e.g.: a custom endpoint calling refresh manually but returning a 401 may cause issues with anti-csrf, but I'm not a 100% sure about it.